### PR TITLE
Refactor towards multi-sample support

### PR DESCRIPTION
**Pull request by fritzo**
@jglidden This also cleans up some of the query tests, in particular using context managers when starting servers. What do you think of the `loom.query.get_server` helper? I'd like to eventually start servers with `root` variables.

### CODE REVIEW

File: loom/query.py [hunk 1]
```diff
@@ -25,16 +25,17 @@
 # TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 # USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-import loom.runner
```
Comment (fritzo) [inline loom/query.py:28]: I've been trying to consistently order imports from most general to most specific

File: loom/query.py [hunk 2]
```diff
@@ -66,11 +67,11 @@ def even_unif_multinomial(total_count, num_choices):
     '''
     quotient = int(total_count / num_choices)
     remainder = total_count - quotient * num_choices
-    result = np.ones((num_choices,), dtype=int) * quotient
```
Comment (fritzo) [inline loom/query.py:69]: I'd like to avoid `import ... as` slang in this repo

File: loom/query.py [hunk 3]
```diff
@@ -352,3 +353,31 @@ def __enter__(self):
 
     def __exit__(self, *unused):
         self.close()
+
+
+def get_protobuf_server(root, debug=False, profile=None):
```
Comment (jglidden) [inline loom/query.py:358]: This is a big improvement.

**Comment by jglidden** [comment]
+1

**Pull request merged**